### DISCUSSION TITLE
what is interest_rate?

### DISCUSSION THREAD

**Discussion by rezunli96** [yasserfarouk/scml#17] [standard-qa] [answered]
In scml2020/world.py the simulation_step() function:
```
    def simulation_step(self, stage):
        self._registered_negs: Dict[Tuple[str], int] = Counter()
        s = self.current_step
        if stage == 0:
            # pay interests for negative balances
            # -----------------------------------
            if self.interest_rate > 0.0:
                for agent, factory, _ in self.afp:
                    if factory.current_balance < 0 and not factory.is_bankrupt:
                        to_pay = -int(
                            math.ceil(self.interest_rate * factory.current_balance)
                        )
                        factory.pay(to_pay)
```
There is a variable interest_rate that seems not to be specified in the document. What is its role in the game?
Also in this code when will a factory have negative balance but yet not claimed to be bankrupt? According to the document this factory will be forced to sell its current products on spot market trying to get back to positive balance right?

**Comment by yasserfarouk** [comment, marked as answer]
The implementation is more general than the game description for future proofing. SCMLWorld can allow agents to borrow from a virtual bank in principle at an interest rate. Nevertheless, this is **always** disabled. The behavior in the game description is guaranteed. Once the agent’s balance gets negative it is declared bankrupt **immediately** and gets liquidated.
